adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -185,42 +183,44 @@ boolean isRemotelySettled() {
 
     //---------------------------------------------< private methods >---
     /**
-     * Creates an ErrorCondition using the given service invocation error to provide an error condition value
-     * and description.
+     * Creates an ErrorCondition using the given throwable to provide an error condition value and
+     * description. All throwables that are not service invocation exceptions will be mapped to {@link AmqpError#PRECONDITION_FAILED}.
      *
-     * @param error The service invocation error.
+     * @param t The throwable to map to an error condition.
      * @return The ErrorCondition.
      */
-    static ErrorCondition getErrorCondition(final ServiceInvocationException error) {
-        switch (error.getErrorCode()) {
-        case HttpURLConnection.HTTP_BAD_REQUEST:
-            return ProtonHelper.condition(Constants.AMQP_BAD_REQUEST, error.getMessage());
-        case HttpURLConnection.HTTP_FORBIDDEN:
-            return ProtonHelper.condition(AmqpError.UNAUTHORIZED_ACCESS, error.getMessage());
-        default:
-            return ProtonHelper.condition(AmqpError.PRECONDITION_FAILED, error.getMessage());
+    static ErrorCondition getErrorCondition(final Throwable t) {
+        if (ServiceInvocationException.class.isInstance(t)) {
+            final ServiceInvocationException error = (ServiceInvocationException) t;
+            switch (error.getErrorCode()) {
+            case HttpURLConnection.HTTP_BAD_REQUEST:
+                return ProtonHelper.condition(Constants.AMQP_BAD_REQUEST, error.getMessage());
+            case HttpURLConnection.HTTP_FORBIDDEN:
+                return ProtonHelper.condition(AmqpError.UNAUTHORIZED_ACCESS, error.getMessage());
+            default:
+                return ProtonHelper.condition(AmqpError.PRECONDITION_FAILED, error.getMessage());
+            }
+        } else {
+            return ProtonHelper.condition(AmqpError.PRECONDITION_FAILED, t.getMessage());
         }
     }
 
     /**
-     * Validates the address contained in an AMQP 1.0 message property.
+     * Validates the address contained in an AMQP 1.0 message.
      * 
-     * @param to The property to validate.
+     * @param address The message address to validate.
      * @return A future indicating the outcome of the validation operation.
+     * @throws ServiceInvocationException if the message address is not valid.
      */
-    Future<Void> validateMessageProperty(final String to) {
-        final Future<Void> propertyCheck = Future.future();
-        if (to == null || to.isEmpty()) {
-            propertyCheck.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "Message _to propery cannot be null or empty"));
+    void validateMessageAddress(final String address) throws ServiceInvocationException {
```

Comment:
is there a reason for this method to be package private instead of private?
I am asking because if this is package private then the JavaDocs are basically useless because they do not give any clue regarding the validation that is actually taking place. In particular, the first null check only makes sense in the context of the target address being `null` as well, right? So, for this method to return a reasonable result, you actually need to have done some checks before as well, right? This would probably be fine, though, if the method was private, so that it cannot be invoked out of the context of this class.

Reply:
sorry, this method was intended to be a private method of the class